python/2_add_two_numbers.py

```diff
@@ -0,0 +1,60 @@
+from typing import Optional
+
+
+# Definition for singly-linked list.
+class ListNode:
+    def __init__(self, val=0, next=None):
+        self.val = val
+        self.next = next
+
+
+class Solution:
+    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
+        """Add two numbers represented by linked lists.
+
+        Each linked list stores a non-negative integer in reverse order, where each node
+        contains a single digit. The task is to return the sum as a new linked list,
+        also in reverse order.
+
+        Example:
+            Input:
+                l1 = (2 -> 4 -> 3)   # represents 342
+                l2 = (5 -> 6 -> 4)   # represents 465
+            Output:
+                (7 -> 0 -> 8)        # represents 807
+
+        Algorithm:
+            - Traverse both linked lists simultaneously.
+            - At each step, add corresponding digits and carry from the previous step.
+            - Create a new node for the digit (sum % 10).
+            - Update carry = sum // 10.
+            - Continue until both lists are fully processed and no carry remains.
+
+        Args:
+            l1 (Optional[ListNode]): Head of the first linked list.
+            l2 (Optional[ListNode]): Head of the second linked list.
+
+        Returns:
+            Optional[ListNode]: Head of the resulting linked list representing the sum.
+
+        Notes:
+            - Time complexity: O(max(m, n)), where m and n are the lengths of l1 and l2.
+            - Space complexity: O(max(m, n)), since a new linked list is created.
+        """
+        carry_up = 0
+        head = ListNode()
+        node = head
+        while l1 or l2 or carry_up == 1:
+            added = (l1.val if l1 else 0) + (l2.val if l2 else 0) + carry_up
```

Comment:
For the sake of readability, could you please rewrite this ternary operator as a standard if/else statement?